### DISCUSSION TITLE
Limit to number of entities that are returned when enumerating files in a folder?

### DISCUSSION THREAD

**Discussion by Amwins-Bass-Guy** [pnp/pnpcore#1451] [Q&A]
Does anyone know whether there is a limit when enumerating files in a given folder?  I am concerned about the eventuality of having several thousand files in a folder and running into issues with all files not being enumerated and hampering the selection of a requested file (by ListItemId):

```c#
var folder= await pnpContext.Web.GetFolderByServerRelativeUrlAsync($"{siteUrl}/{parentId}/{childId}",
            f => f.Exists,
            f => f.Files.QueryProperties(
                fi => fi.ListItemAllFields.QueryProperties(li => li.Id, li => li.UniqueId,
                    li => li.ContentType.QueryProperties(ct => ct.Name, ct => ct.Id))
            ));
            
 var requestedDocument = folder.Files
            .AsRequested()
            .Where(f => f.ListItemAllFields is not null)
            .FirstOrDefault(f => f.ListItemAllFields.Id == someRequestedListItemId);
```

**Comment by jansenbe** [comment]
@Amwins-Bass-Guy : loading files using above approach will not load all files, if you use above code on a large library (+5000 files) you'll most likely see this error "The attempted operation is prohibited because it exceeds the list view threshold".  See https://github.com/pnp/pnpcore/discussions/586 for an approach that help you
